src/controllers/admin/events.js

```diff
@@ -10,26 +10,31 @@ var eventsController = module.exports;
 
 eventsController.get = function (req, res, next) {
 	var page = parseInt(req.query.page, 10) || 1;
-	var itemsPerPage = 20;
+	var itemsPerPage = parseInt(req.query.perPage, 10) || 20;
 	var start = (page - 1) * itemsPerPage;
 	var stop = start + itemsPerPage - 1;
 
-	var currentFilter = req.query.filter || '';
+	// Limit by date
+	var from = req.query.start ? new Date(req.query.start) || undefined : undefined;
+	var to = req.query.end ? new Date(req.query.end) || undefined : new Date();
+	from = from && from.setHours(0, 0, 0, 0);
+	to = to && to.setHours(23, 59, 59, 999);
+
+	var currentFilter = req.query.type || '';
 
 	async.waterfall([
 		function (next) {
 			async.parallel({
 				eventCount: function (next) {
-					db.sortedSetCard('events:time' + (currentFilter ? ':' + currentFilter : ''), next);
+					db.sortedSetCount('events:time' + (currentFilter ? ':' + currentFilter : ''), from || '-inf', to, next);
```

Comment:
need `from.getTime()`?

Reply:
>The setHours() method sets the hours for a specified date according to local time, and returns the number of milliseconds since January 1, 1970 00:00:00 UTC until the time represented by the updated Date instance.

A little confusing, that it returns the unix time, but no issues here. An explicit explanation via comment might be helpful though, because that threw me for a loop too.